### PR TITLE
Frame inversion

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/geometry/frame.py [hunk 1]
```diff
@@ -604,6 +604,18 @@ def to_transformation(self):
     # Methods
     # ==========================================================================
 
+    def invert(self):
```
Comment (chenkasirer) [inline src/compas/geometry/frame.py:607]: not sure but should this be maybe called `flip`?

File: src/compas/geometry/frame.py [hunk 2]
```diff
@@ -604,6 +604,18 @@ def to_transformation(self):
...
+        """Invert the frame while keeping the X axis fixed."""
+        self._yaxis = self.yaxis * -1
+        self._zaxis = None
+
+    def inverted(self):
+        """Return an inverted copy of the frame."""
+        frame = self.copy()  # type: Frame
+        frame.invert()
+        print(frame.xaxis, frame.yaxis, frame.zaxis)
```
Comment (chenkasirer) [inline src/compas/geometry/frame.py]: remove `print`?

**Review by chenkasirer — APPROVED** [review]
lgtm
just a couple of small comments

File: tests/compas/geometry/test_frame.py [hunk 3]
```diff
@@ -92,3 +92,35 @@ def test_interpolate_frame_start_end():
 
     three_quarter_frame = frame1.interpolate_frame(frame2, 0.75)
     assert TOL.is_allclose([math.degrees(three_quarter_frame.axis_angle_vector.y)], [-67.5], atol=TOL.angular)
+
+
+def test_frame_invert():
+    frame = Frame([0, 0, 0])
+
+    assert TOL.is_close(frame.xaxis.dot([1, 0, 0]), 1.0)
```
Comment (chenkasirer) [inline tests/compas/geometry/test_frame.py:100]: just out of curiosity what's the advantage of this over `TOL.is_allclose(frame.xaxis, [1, 0, 0])`

File: src/compas/geometry/frame.py (hunk 1 above)
Comment (tomvanmele) [reply src/compas/geometry/frame.py:607]: in `Vector` it is also `invert`, but we can add an alias in both perhaps

File: tests/compas/geometry/test_frame.py (hunk 3 above)
Comment (tomvanmele) [reply tests/compas/geometry/test_frame.py:100]: no advantage. am just used to expressing this using dot

**Pull request merged**